### ISSUE TITLE
Symlink Handling

### ISSUE THREAD

**Issue by fbrnc** [fbrnc/devbox-cookbook#1]
In [this tweet](https://twitter.com/fschmengler/status/661448211786285056) @schmengler is asking why I'm disabling symlinks in the Samba configuration:
https://github.com/fbrnc/devbox-cookbook/blob/master/shares.json#L8

I never thought about how enabling this could boost the performance (and I also didn't try this now), but the main reason I disabled this in the first place was because it's very convenient to NOT expose symlinked files to my host system since these files are either duplicates (using modman you'd end up having the files in the `.modman` dir and in the Magento core) or - in my setup - the `media` and `var` directories that I don't want in PHPStorm.

So disabling Sylinks on the Samba level allows me to simply add the complete project root to PHPStorm without having to worry about stuff I don't want in PHPStorm.

I never experienced performance issues with Samba (but I hear others do), but if this helps it might be worth enabling it and instead selectivly add directories to the PHPStorm project we care about.

I'm creating this as a GitHub issue since this wouldn't fit in a tweet and I'm sincerely interested in your opinion on that and how you solve the "files-live-in-guest-but-I-want-to-edit-them-on-my-host" problem.

**Comment by schmengler** [comment]
I don't know about performance, the main issue for me was efficient usage of development tools. The Magicento plugin searches for modules in the Magento root directory (for autocompletion etc.) and will only find my modules if the symlinks are being followed. The alternative would be to configure an additional Magento root in Magicento for each module, which is possible but tedious.

Duplicate files in the PHPStorm index _are_ an issue but I am willing to live with that. I tried to mark ".modman" to be excluded from source but unfortunately that made the path mappings for XDebug stop working.

If anybody has a different solution, I'm interested as well, since mine still seems not to be the best one.

**Comment by fbrnc** [comment]
Good point. 
I bought the Magicento plugin a while ago. Used it a couple of days and really liked it and then ended up not using it anymore. Not sure why...
But also as you can probably tell I was only using features that don't expect any files to be in a certain place.
So, yes if that's important for you you probably should enable symlinks, but I can also see how the duplicate files will be confusing during development.

Magento 2 is close and then we won't have to deal with symlinks anymore :)

**Comment by schmengler** [comment]
> Magento 2 is close and then we won't have to deal with symlinks anymore :)

Except us poor people that will probably maintain Magento 1 shops for the next 5+ years. But still: YAY :)

**Comment by fbrnc** [comment]
Yes, I'm pretty sure we'll also be doing Magento 1 for another while...

Sadly syncing data between the host and the guest continues to be a tricky problem (and for all people who like using PHPStorms sync instead of Samba or something else that's transparent, they will run into some problems with all the auto-generated files not being available to PHPStorm unless you manually sync, while debugging).

Another option could be to install PHPStorm inside the client. But of course that would require a lot more overhead in the box. You could then access PHPStorm either via VirtualBoxes "blended" mode (not sure what this is actually called), or even better you could run a x server (e.g.  xming) and render PHPStorm on your host.
Not sure if this feels snappy and looks good, but might be worth trying.
